### PR TITLE
Adding optional fix for monitors with shitty colour display, or low gamma

### PR DESCRIPTION
**Pull request by jgh713**
AKA pretty colours for people with bad eyesight or shitty equipment.

### CODE REVIEW

**Comment by jgh713** [comment]
Adding a built-in method by which to change the echo colour from that god-awful nigh-unreadable blue that everyone seems to be used to. Defaults to the typical disgusting echolink blue. If someone wants to take this and turn it into a proper mconfig, that would be nice too, but for now this just offers an alternative to that eye-straining nonsense that echolink defaults to.

File: mudlet-mapper.xml [hunk 1]
```diff
@@ -222,40 +222,38 @@ end</script>
             <Trigger isActive="yes" isFolder="no" isTempTrigger="no" isMultiline="no" isPerlSlashGOption="no" isColorizerTrigger="no" isFilterTrigger="no" isSoundTrigger="no" isColorTrigger="no" isColorTriggerFg="no" isColorTriggerBg="no">
                 <name>Alertness (Achaea+Imperian)</name>
                 <script>mmp.alertness = mmp.alertness or {}
-
+cecho(&quot;&lt;red&gt;\n&quot;..matches[4])
```
Comment (keneanung) [inline mudlet-mapper.xml]: This looks like a debug echo (and shouldn't be here), no?

File: mudlet-mapper.xml [hunk 2]
```diff
@@ -222,40 +222,38 @@ end</script>
...
-
-if matches[3] == currentroomname then
-  alertness.here = alertness.here or {}
-  alertness.here[#alertness.here+1] = matches[2]
-  deleteLine()
+ 
+ if matches[3] == currentroomname then
+ alertness.here = alertness.here or {}
+ alertness.here[#alertness.here+1] = matches[2]
+ deleteLine()
```
Comment (keneanung) [inline mudlet-mapper.xml]: Could you improve the indenting here?

File: mudlet-mapper.xml (hunk 1 above)
Comment (QwindorRousseau) [inline mudlet-mapper.xml]: Uh yea totally forgot about that. Will edit it once I can figure out github again.

File: mudlet-mapper.xml (hunk 2 above)
Comment (QwindorRousseau) [inline mudlet-mapper.xml]: Yea let me redo it again here

Comment (QwindorRousseau) [inline mudlet-mapper.xml]: Considering @Ahmet already has it set up for the pull I sent the revised script to him to adjust. Easier that way considering I'm still trying to figure out how to work github stuff

**Comment by jgh713** [comment]
Fixed up the alertness stuff. Thanks @QwindorRousseau 

Should be all good.

**Comment by jgh713** [comment]
My local git went kerbloom so I just rebased and re-added the changes. Dirty, but easy. Github is taking its sweet time to recognize things have actually changed, but there should be no issues.

File: mudlet-mapper.xml [hunk 3]
```diff
@@ -1355,6 +1348,45 @@ mmp.locateAndEcho(room)</script>
...
+                    <script>local dirnums = mmp.getnums(matches[4], true)
+local which
+
+if dirnums == nil then
+	echo(&quot; (?)&quot;)
+else
+	if #dirnums == 1 then
+		which = dirnums[1]
+		--echo(string.rep(&quot; &quot;, 30 - #matches[4]))
+		--cechoLink(&quot; - &lt;&quot;..mmp.echo_colour..&quot;&gt;&quot;..which, [[gotoRoom(]]..which..[[)]], &quot;Travel to &quot;..which, true)
```
Comment (keneanung) [inline mudlet-mapper.xml]: Debug echoes?

File: mudlet-mapper.xml [hunk 4]
```diff
@@ -1355,6 +1348,45 @@ mmp.locateAndEcho(room)</script>
...
+   	else
+		for index, val in ipairs(dirnums) do
+			if getRoomArea(val) == getRoomArea(mmp.currentroom) then which = val end
+		end
+		if which == nil then which = dirnums[1] end
+		--echo(string.rep(&quot; &quot;, 35 - #matches[4]))
+		--echo(&quot; - &quot;) mmp.echonums(matches[4])
```
Comment (keneanung) [inline mudlet-mapper.xml]: Debug echoes?

File: mudlet-mapper.xml [hunk 5]
```diff
@@ -1355,6 +1348,45 @@ mmp.locateAndEcho(room)</script>
...
+else
+	if #dirnums == 1 then
+		which = dirnums[1]
+		--echo(string.rep(&quot; &quot;, 30 - #matches[4]))
+		--cechoLink(&quot; - &lt;&quot;..mmp.echo_colour..&quot;&gt;&quot;..which, [[gotoRoom(]]..which..[[)]], &quot;Travel to &quot;..which, true)
+   	else
+		for index, val in ipairs(dirnums) do
+			if getRoomArea(val) == getRoomArea(mmp.currentroom) then which = val end
+		end
+		if which == nil then which = dirnums[1] end
```
Comment (keneanung) [inline mudlet-mapper.xml]: I'm not sure this should be the preferred solution. Your solution will
1. suggest going to the last room with that name in the area. Maybe you can try and filter for a room with the `shop` mark for the directory listing. But even then it's not necessarily unique and I'd inform the user about that.
2. If no room is found in the same area, then it suggests going to the first one in the list, no matter where that is. And no user can find out, where it'll take them if you give no indication. I'd also inform the user that no room in the current area is found.

**Comment by jgh713** [comment]
Not debug echoes. Originally had it echoing the room numbers off to the side but that looked horrid, so I knocked those out for now.

#1 (i forgot a break, but first/last doesnt matter because) I ran a few checks. May change at some point in the future (though I doubt it), or I may have missed something, but as far as I could tell there are multiple shops with the same room name, but no two reside within the same area (and there haven't been since Shallam), thus filtering by area instead of shop marks.

#2 Tooltips are a thing, so it's not completely impossible, but I see what you mean. Thinking an echo off to the side would be best for something that (should) be a rare case.

**Comment by keneanung** [comment]
But the echoes are nevertheless commented out and thus shouldn't be there :stuck_out_tongue: 

1: It doesn't matter for now. But shop rooms are player assigned names, so it may happen that the same names occur. Or Rurin shops on generic rooms?

**Comment by vadi2** [comment]
Can you make a PR just about the colour changes only? Otherwise it'll take a while to get in with all the other changes you're sneaking in.

**Comment by jgh713** [comment]
Removed the directory triggers, but left the bugfixes that seem to be relatively simple (read: quick to review and obvious fixes). Will refrain from the other stuff I'm doing upstream for now :P

File: mudlet-mapper.xml [hunk 6]
```diff
@@ -5022,7 +5016,7 @@ function mmp.openDoor()
   -- not going anywhere? don't do anything
   if not mmp.speedWalkDir[speedWalkCounter] then return end
 
-  send(&quot;open door &quot; .. mmp.speedWalkDir[speedWalkCounter]:gsub(&quot;sprint &quot;, &quot;&quot;):gsub(&quot;dash &quot;, &quot;&quot;), false)
+  send(&quot;open door &quot; .. mmp.speedWalkDir[speedWalkCounter]:gsub(&quot;sprint &quot;, &quot;&quot;):gsub(&quot;dash &quot;, &quot;&quot;):gsub(&quot;gallop &quot;, &quot;&quot;), false)
```
Comment (keneanung) [inline mudlet-mapper.xml]: There's also Jester runaway missing here.

File: mudlet-mapper.xml [hunk 7]
```diff
@@ -5002,7 +4996,7 @@ function mmp.swim()
   -- not going anywhere? don't do anything
   if not mmp.speedWalkDir[speedWalkCounter] then return end
 
-  send(&quot;swim &quot; .. mmp.speedWalkDir[speedWalkCounter], false)
+  send(&quot;swim &quot; .. mmp.speedWalkDir[speedWalkCounter]:gsub(&quot;sprint &quot;, &quot;&quot;):gsub(&quot;dash &quot;, &quot;&quot;):gsub(&quot;gallop &quot;, &quot;&quot;), false)
```
Comment (keneanung) [inline mudlet-mapper.xml]: Wouldn't this make the mapper think it's gone off course (because it expects the next room number to be the dash end room)?

**Comment by jgh713** [comment]
It would, yes! The problem is, you can't swim sprint!

**Comment by keneanung** [comment]
You can't. But what I'm suggesting is that the proposed solution might be too short sighted (without having analyzed the whole path handling code). I'd be happy to be shown wrong /vetoed here.

**Comment by jgh713** [comment]
Perhaps! From the testing I've done, it adds negligible time to the pathing process, especially when you take into account the time it takes to actually swim forward. Not to mention how rare it is that someone will have sprinting/dashing/galloping/whatever and not waterwalking. Shouldn't actually be possible if they're on a mount. It may be a shortsighted solution, but it's one that works, and is better than the user's mapper just flailing about uselessly, in my opinion.

**Comment by keneanung** [comment]
I wasn't really worried about the time overhead, but the confusing output towards the user. It's like "dash", "walk", "walk", "swim" - "Oh Hey, we went off path", "swim", "dash"

**Comment by jgh713** [comment]
Dashing already has quirks like that. Sprinting into doors, or rubble (which also locks up while pathing through dash, if I recall correctly). Dunno. If you feel like taking a deeper look, or setting it aside for now, that's fine. Just a temporary patch to fix the nuisance of things not working properly.

**Comment by keneanung** [comment]
Maybe @vadi2 can decide on that?

**Comment by vadi2** [comment]
Agreed it's not nice, but it's faster, it's better than the slower and already not nice. Think most people just want to get from A to B the quickest.

+1 on the PR, the colours change looks fine. Thank you for your contribution!

**Comment by keneanung** [comment]
Perfect, now only the jester runaway is missing in the `gsub`s :stuck_out_tongue: @jgh713

**Comment by keneanung** [comment]
Thank you very much :)

**Pull request merged**